### PR TITLE
[IMPROVE] Typing indicators now use Real Names

### LINKED ISSUE #11145: User IDs still showing as "[user id] is typing..." and elsewhere even with "Use Real Name" setting on
### Description:

After setting the "Use Real Name" setting on under Administration > Layout > User Interface, there are still places where the user ID, rather than the name is being displayed:

These include:

* When a user in a channel is typing
* When a user makes changes in a channel, e.g. sets the topic

### Steps to reproduce:

1. Enable the "Use Real Name" setting
2. Note that the user names are now displayed in e.g. mentions and the side panel
3. Make a change in a room, like setting the topic; the room history shows the user ID instead of name
4. When a user in a room is typing, it shows the user ID instead of name

### Expected behavior:

When making the "Use Real Name" setting, it should apply everywhere where user IDs normally appear.

### Actual behavior:

User IDs still show up in various places.

### Server Setup Information:

- Version of Rocket.Chat Server: 0.65.1
- Operating System: 4.15.0-23-generic
- Deployment Method: snap
- Number of Running Instances:  1
- DB Replicaset Oplog: 
- NodeJS Version: v8.11.2
- MongoDB Version:

### PR DESCRIPTION
**Pull request by vynmera**
Part of an attempt to improve #11145

If the admin has "Use Real Name" on, typing indicators now show the real name, making life a lot easier for those who auto-generate usernames or otherwise have usernames that don't correlate with the user.

**Before, or with "Use Real Name" off**

**After, with "Use Real Name" on**

Enjoy!

### CODE REVIEW

File: packages/rocketchat-lib/server/functions/Notifications.js [hunk 1]
```diff
@@ -134,10 +139,11 @@ RocketChat.Notifications.streamRoom.allowWrite(function(eventName, username, typ
 
 		const user = Meteor.users.findOne(this.userId, {
 			fields: {
-				username: 1
+				username: 1,
+				name: 1
 			}
 		});
-		if (user != null && user.username === username) {
+		if (user != null && (user.username === username || (useRealNames() && (user.name === username)))) {
```
Comment (karlprieb) [inline packages/rocketchat-lib/server/functions/Notifications.js]: Why did you create a function called `userRealNames` if you use only one time? Why no put `RocketChat.settings.get('UI_Use_Real_Name')` directly on this if instead?

File: packages/rocketchat-ui/client/lib/msgTyping.js [hunk 2]
```diff
@@ -9,14 +9,25 @@ export const MsgTyping = (function() {
 	const usersTyping = {};
 	const dep = new Tracker.Dependency;
 
+	const useRealNames = function() {
+		return RocketChat.settings.get('UI_Use_Real_Name');
+	};
+
+	const shownName = function(user) {
+		if (useRealNames()) {
```
Comment (karlprieb) [inline packages/rocketchat-ui/client/lib/msgTyping.js]: Same here, just use `RocketChat.settings.get('UI_Use_Real_Name')`

File: packages/rocketchat-ui/client/lib/msgTyping.js [hunk 3]
```diff
@@ -43,7 +50,7 @@ export const MsgTyping = (function() {
 			timeouts[room] = null;
 		}
 		const user = Meteor.user();
-		return RocketChat.Notifications.notifyRoom(room, 'typing', user && user.username, false);
+		return RocketChat.Notifications.notifyRoom(room, 'typing', shownName(user), false);
```
Comment (mrsimpson) [inline packages/rocketchat-ui/client/lib/msgTyping.js]: @vynmera this broke the notifications in some cases.
Sometimes, `user` is missing completely - I believe if nobody is typing.
The earlier `user && ...` would evaluate to false, not notifying the stream - this was intended. Please restore this before merging

**Comment by vynmera** [comment]
@mrsimpson Should be fixed :)

**Pull request merged**